docs/architecture/go_components.md

```diff
@@ -0,0 +1,117 @@
+	go.uber.org/multierr v1.6.0 // indirect
+	golang.org/x/net v0.0.0-20210226172049-e18ecbb05110
+	k8s.io/api v0.20.4
+	modernc.org/sqlite v1.10.8
+```
+
+Notice the variety in paths? Google allows "vanity" package names (like `go.uber.org/multierr` - which are automatically redirected to the source by Google's proxy service.
+
+## Constructing coordinates
+
+In order to request a definition of a go component from ClearlyDefined (which will also queue up a harvest if ClearlyDefined does not have a definition for the component), a user will need to construct coordinates in the API call.
+
+### For a Maven component
+
+For example, in order to request a maven package, the API call would look like this:
+
+```bash
+curl -X GET https://clearlydefined.io/definitions/maven/mavencentral/com.apollographql.apollo3/apollo-runtime-jvm/3.0.0-dev14" -H "accept: */*"
+```
+
+In this case, the coordinates are maven/mavencentral/com.apollographql.apollo3/apollo-runtime-jvm/3.0.0-dev14
+
+Let's break this down:
+* **maven** is the **type** 
+* **mavencentral** is the **provider**
+* **com.apollographql.apollo3** is the **namespace**
+* **apollo-runtime-jvm** is the **name**
+* **3.0.0-dev14** is the **revision**
+
+When we run this API call:
+
+```bash
+curl -X GET https://clearlydefined.io/definitions/maven/mavencentral/com.apollographql.apollo3/apollo-runtime-jvm/3.0.0-dev14" -H "accept: */*"
+```
+
+It harvests the component from [here](https://repo1.maven.org/maven2/com/apollographql/apollo3/apollo-runtime-jvm/3.0.0-dev14/).
+
+Notice something in the url - the namespace is "com/apollographql/apollo3" - there are a lot of slashes in there. 
+
+The way we translate this namespace into coordinates is to replace the slashes in the namespace with periods.
+
+So "com/apollographql/apollo3" becomes "com.apollographql.apollo3".
+
+### For a Go component
+
+A Go component's coordinates will also need to consist of a type, provider, namespace, name, and revision.
+
+Go's support of vanity package names makes this a little different than other components. While there are a few different approaches we could take, we should map the coordinates to what go developers know the package as.
+
+* **type** should be go
+* **provider** should be "proxy.golang.org" (we may need to add other providers in the future)
+* **namespace** should be anything that appears before the package name
+* **name** should be the name of the package
+* **revision** should be the component revision (note - go components seem to have multiple ways of specifying revisons - sometimes version numbers, sometimes commit hashes)
+
+Like with Maven components, we translate namespaces that have multiple slashes (like `golang.org/x`) into namepaces with periods (`golang.org.x`).
+
+For example: `go.uber.org/multierr v1.6.0` maps to `go/proxy.golang.org/go.uber.org/multierr/v1.6.0`
+
+**Examples that seem to work well:**
+
+* `collectd.org v0.5.0` maps to `go/proxy.golang.org/collectd.org/v0.5.0`
```

Comment:
There was some previous discussion of formatting the coordinates with the namespace as the provider - i.e. `go.uber.org/multierr v1.6.0` would map to `go/go.uber.org/-/multierr/v1.6.0`. However, this wouldn't work for packages that are imported as a single name - like `collectd.org v0.5.0` which would map to `go/-/-/collectd.org/v0.5.0`. This won't work unless we have some sort of default for the provider in cases like these.

Reply:
Sorry, what is the current proposal for package names with no / ? Should it be:
`collectd.org v0.5.0` maps to `go/proxy.golang.org/-/collectd.org/v0.5.0`

Reply:
ah! That's a type on my part. `go/proxy.golang.org/-/collectd.org/v0.5.0` is correct!

Reply:
Fixed!